### ISSUE TITLE
Can some of you post links to some of the visualizations you're created?

### ISSUE THREAD

**Issue by mwherman2000** [Microsoft/automatic-graph-layout#74]
That is, create a visual gallery of the types of real/demo visualizations you've created?

Thank you,
Michael Herman (Toronto)

**Comment by vdisasm** [comment]
Well, that's one of my latest graphs. It is control flow graph built from x86 assembly.
Using Sugiyama layout and rectilinear edge routing.

![Control Flow Graph](https://3.bp.blogspot.com/-nmSZ4ZpSMk0/Vyx2IDYbnJI/AAAAAAAAADc/78wZITvlzrUbHt2SVhKojTIun4Zdpd1ygCLcB/s1600/05-after-if-then-foldings.png)

Few simpler ones:
![CFG entry](https://1.bp.blogspot.com/-KC5fBrmrw8w/VxT34jFERzI/AAAAAAAAADI/J6cukr90oz8WIjaM_HgKmBX61Lokwok0wCLcB/s1600/02-start.png)
![CFG branches](https://3.bp.blogspot.com/-NHhPApM3c5w/VxT35MrdciI/AAAAAAAAADM/jNKfZT12b1E4HRyPrqHMOlY2aR5i0FlRQCLcB/s1600/04-switch.png)

From:
http://blog.vdisasm.com/2016/05/folding-if-then-else-structures.html
http://blog.vdisasm.com/2016/04/code-analyzer-graphs.html

**Comment by JTealdi** [comment]
Here is my Customer Care WorkFlow editor (wip).
Some graphs have over 500 nodes. This one is the smallest, with just 20.

![untitled](https://cloud.githubusercontent.com/assets/20539806/16953193/8500bb96-4da2-11e6-8656-8ff214b10c99.png)

**Comment by mwherman2000** [comment]
Thank you for posting these graphs. 

JTealdi, have you considered using BPMN or ArchiMate shapes for your workflow editor?

**Comment by JTealdi** [comment]
Yep. Eventually I´ll modify my DrawNodeDelegate further to add new shapes. As it is, it only handles fillcolor to make a gradient.

The old workflows were designed in Visio, so I'm aiming to recreate some of the shapes they're using.
I don't have much experience with Drawing.Graphics, so any advice/example code is welcome :D

**Comment by K2** [comment]
This is a soon to be released tool (thanks for fixing that triangle bug ;) that I want use graphmaps to find relationships between executing threads by and other hard to determine relationships that might lead to security issues (TOCTOU, or race conditions).

![image](https://user-images.githubusercontent.com/16319957/29814459-386e8dc6-8c62-11e7-8e99-cf014aa46df1.png)

**Comment by smwikipedia** [comment]
This is the graph I generated for the UEFI codebase.
![untitled](https://user-images.githubusercontent.com/999318/30897398-99935626-a388-11e7-9ac4-5f7946711501.png)

**Comment by pburgepally** [comment]
> This is a soon to be released tool (thanks for fixing that triangle bug ;) that I want use graphmaps to find relationships between executing threads by and other hard to determine relationships that might lead to security issues (TOCTOU, or race conditions).
> 
> ![image](https://user-images.githubusercontent.com/16319957/29814459-386e8dc6-8c62-11e7-8e99-cf014aa46df1.png)

Is this using web MSAGL or the one for winforms.

**Comment by levnach** [comment]
Graphmaps is build on top of WPF.
